nginx.conf.template

```diff
@@ -1,51 +1,76 @@
-# /etc/nginx/nginx.conf.template
-
+# /etc/nginx/nginx.conf
 user nginx;
-
-# Sets the number of worker processes based on CPU cores available
-worker_processes auto; 
+worker_processes auto; # Adjust based on the number of CPU cores
 pid /var/run/nginx.pid;
 
 events {
-    # Handles how many connections each worker process can handle simultaneously
-    worker_connections 1024; 
+    worker_connections 1024; # Maximum simultaneous connections per worker
 }
 
-# Handles requests for the Vue static files
 http {
-    # Loads the mime types file that defines mappings between file extensions and content types
-    include /etc/nginx/mime.types; 
-
-    # Default content type if unspecified
-    default_type application/octet-stream;
+    # Load standard MIME types
+    include /etc/nginx/mime.types;
+    default_type application/octet-stream; # Fallback MIME type for unknown files
 
-    # Enables the sendfile system call for faster file transfers
+    # Optimize file transfer
     sendfile on;
-
-    # Time in seconds to keep a connection open after completing a request
     keepalive_timeout 65;
 
+    # Define allowed CORS origins
+    map $http_origin $cors_origin {
```

Comment:
Just put:
```
    # # Define allowed CORS origins
    map $http_origin $cors_origin {
        default ""; # Default to disallow all origins
    }
```
    
    Don't need to specify extra lines there

Reply:
I get this error message when requesting the data if I do not specify who is allowed to access the data. 
![image](https://github.com/user-attachments/assets/7af40c7c-a38e-4864-92c4-1e58f6a47aa3)

Reply:
ah it makes sense. I don't have a test.csv so I don't see the problem.
You may want to add a comment there in that block to explain why.
Looks like you're targeting localhost environment at 5173 and 8080 (very specific fixed ports number, so will it work for all the cases????)